### DISCUSSION TITLE
[API Request] Extension to signify that a server supports playing media other than songs in jukebox mode

### DISCUSSION THREAD

**Discussion by dweymouth** [opensubsonic/open-subsonic-api#86] [API requests]
### Type of change

API extension

### Proposal description

The current spec for `jukeboxControl` says that song IDs are allowed, but makes no mention of podcastEpisode or internetRadioStation IDs. And there is no guarantee that IDs across media types are unique, so clients cannot assume any IDs other than song IDs are supported by jukeboxControl.

We should add an extension to indicate a server can accept IDs for all playable media (excluding video: that is currently songs, podcast episodes, and radio stations, but this extension should cover any possible future audio media types eg audiobooks) for the `jukeboxControl` endpoint. For non-seekable items like radio stations, the server should be able to ignore the `offset` parameter if passed.

gonic already supports this as of today, at least for radio stations - https://github.com/sentriz/gonic/issues/481 - and DSub also supports passing radio IDs (which is technically an improper use of the base Subsonic API). Navidrome currently *does not* support playing radio stations in Jukebox mode (from a code audit).

### Backward compatibility impact

None

### Backward compatibility

- [x] No backward compatibility impact.

### API details

Proposing a new OpenSubsonic extension called `jukeboxMediaTypes`.

Version 1 of the extension adds two new arguments to the `jukeboxControl` endpoint:
* `podcastEpisodeID`
* `radioStationID`
Like the existing `id` parameter, multiple instances of the parameter are allowed to enqueue multiple items at once. Eg:

`/jukeboxControl?action=add&id=song1&radioStationID=radio1&id=song2&radioStationID=radio2`

**Comment by dweymouth** [comment]
I think this should be a really simple one, it's just the naming of the extension that needs to be decided on.

**Reply by Tolriq** [reply]
We should not encourage using of unnamed Id value as any possible media type. The extension should also add new parameters podcastId, radioId, .... And do the same for the other cases where the id can be anything.

**Reply by dweymouth** [reply]
So your suggestion would be to add one extension that encompasses two new arguments to jukeboxControl: `podcastEpisodeID` and `radioStationID`? That would work for me

**Reply by dweymouth** [reply]
Updated the discussion with a more concrete proposal.

**Reply by Tolriq** [reply]
Yes we must not endorse id without type in OS. For the rest I'm on phone so need to check names another solution would be to add an endpoint to ask for supported media types. Details are up to you that use the API. Just no more id= xxxx param please.

**Comment by dweymouth** [comment]
@sentriz Any thoughts, since gonic supports playing radio stations (and podcasts?) by jukebox? I think you would be the only server that would add support for this extension initially.

**Reply by sentriz** [reply]
sure. gonic will need to support both the typed and untyped ids though since original subsonics supported untyped and internet radio stations
